bin/worker.js

```diff
@@ -18,6 +18,12 @@ export default {
     }
 
     // TODO: Record retrieval stats to D1 asynchronously (do not block response)
-    return await retrieveFile(BASE_URL, pieceCid, env.CACHE_TTL)
+    let response = await retrieveFile(BASE_URL, pieceCid, env.CACHE_TTL)
+    if (response.ok) {
+      response = new Response(response.body, response)
+      response.headers.set('Cache-Control', `max-age=${env.CACHE_TTL}`)
+    }
```

Comment:
```suggestion
    const response = await retrieveFile(BASE_URL, pieceCid, env.CACHE_TTL)
    if (response.ok) {
      response.headers.set('Cache-Control', `max-age=${env.CACHE_TTL}`)
    }
```

Would this work as well?

Reply:
@juliangruber That will not work as response headers are immutable.